widget/window_switcher.lua

```diff
@@ -0,0 +1,400 @@
+local window_switcher_minimized_clients = {} -- The clients that were minimized when the window switcher was activated
+local window_switcher_grabber
+
+local get_num_clients = function(s)
+    local minimized_clients_in_tag = 0
+    local matcher = function (c)
+        return awful.rules.match(c, { minimized = true, skip_taskbar = false, hidden = false, first_tag = s.selected_tag })
+    end
+    for c in awful.client.iterate(matcher) do
+        minimized_clients_in_tag = minimized_clients_in_tag + 1
+    end
+    return minimized_clients_in_tag + #s.clients
+end
+
+local window_switcher_hide = function()
+    -- Add currently focused client to history
+    if client.focus then
+        local window_switcher_last_client = client.focus
+        awful.client.focus.history.add(window_switcher_last_client)
+        -- Raise client that was focused originally
+        -- Then raise last focused client
+        if window_switcher_first_client and window_switcher_first_client.valid then
+            window_switcher_first_client:raise()
+            window_switcher_last_client:raise()
+        end
+    end
+
+    -- Minimize originally minimized clients
+    local s = awful.screen.focused()
+    local clients = s.selected_tag:clients()
+    for _, c in pairs(window_switcher_minimized_clients) do
+        if c and c.valid and not (client.focus and client.focus == c) then
+            c.minimized = true
+        end
+    end
+    -- Reset helper table
+    window_switcher_minimized_clients = {}
+
+    -- Resume recording focus history
+    awful.client.focus.history.enable_tracking()
+    -- Stop and hide window_switcher
+    awful.keygrabber.stop(window_switcher_grabber)
+    s.window_switcher_box.visible = false
+end
+
+local function draw_widget(s, type, client_width, client_height, client_margin, background, border_radius, text_font, font_icons, font_icons_font, mouse_keys)
+    local set_icon = function(item, c)
+        local i = font_icons[c.class] or font_icons['_']
+        item:get_children_by_id('text_icon')[1].markup = "<span foreground='" .. i.color .. "'>" .. i.symbol .. "</span>"
+    end
+
+    local update_thumbnail = function(self, c)
+        local content = gears.surface(c.content)
+        local cr = cairo.Context(content)
+        local x, y, w, h = cr:clip_extents()
+        local img = cairo.ImageSurface.create(cairo.Format.ARGB32, w - x, h - y)
+        cr = cairo.Context(img)
+        cr:set_source_surface(content, 0, 0)
+        cr.operator = cairo.Operator.SOURCE
+        cr:paint()
+        self:get_children_by_id('thumbnail')[1].image = gears.surface.load(img)
+    end
+
+    local icon_widget = function()
+        if (font_icons) ~= nil then
+            return {
+                widget = wibox.widget.textbox,
+                id     = 'text_icon',
+                font   = font_icons_font,
+                forced_width = dpi(50),
+                align  = "center",
```

Comment:
Mixing single and double quotes is a kind of bad manners...

Reply:
It is, but it really depends on the repository you are commiting too, and in this case I don't really care about it... Lol